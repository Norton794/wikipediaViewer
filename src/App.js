import React, { Component } from "react";
import "./App.css";

class App extends Component {
  constructor() {
    super();
    this.state = {
      list: [],
      searchTerm: "React",
    };
    this.onSearchChange = this.onSearchChange.bind(this);
  }

  onSearchChange(e) {
    this.setState({ searchTerm: e.target.value });
  }

  render() {
    const { list, searchTerm } = this.state;
    return (
      <div className="page">
        <h1>WikiViewer</h1>

        <a
          className="random"
          href="https://en.wikipedia.org/wiki/Special:Random"
        >
          Go to a random article
        </a>

        <Search onChange={this.onSearchChange} value={searchTerm}> Search with the Wikipedia API</Search>
        <List list={list}/>
      </div>
    );
  }
}

const Search = ({ value, onChange, children }) => (
  <input
    type="text"
    value={value}
    onChange={onChange}
    placeholder={children}
  />
);

const List = ({ list }) => {
  return (
    <ul>
      {list.map((item) => (
        <li>{item.name}</li>
      ))}
    </ul>
  );
};

export default App;
