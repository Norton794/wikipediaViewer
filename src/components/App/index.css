@import url('https://fonts.googleapis.com/css2?family=STIX+Two+Math&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Quicksand:wght@300&display=swap');
*{
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}
body {
  
  font-family: 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  background-color: #4a6fa5;
  color: #fff;
}

a{
  text-decoration: none;
  color: #000;
}

h1{
  margin-top: 15vh;
  font-family: 'STIX Two Math', serif;
}

a.random{
  color: #fff;
  text-decoration: none;
  font-weight: 700;
  line-height: 16px;
  font-size: 21px;
  margin-top: 15px;
  margin-bottom: 15px;
}



.page{
  display: grid;
  place-items: center;
}

.error{
  margin-top: 50px;
  font-size: 2.1em;
  font-family: 'Quicksand', sans-serif;
}

