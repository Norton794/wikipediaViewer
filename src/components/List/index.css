

/* list */
.cards{
    background-color: #f1f2ee;
    color: #000100;
    margin-top: 20px;
    margin-bottom: 20px;
    padding: 20px 20px;
    cursor: pointer;
    transition: transform 1s;
  }
  
  .cards:hover{
    box-shadow: 7px 7px 7px 7px rgba(0, 0, 0, 0.2);
    
  }
  
  
  .pgTitle{
    text-align: center;
    margin-top: 10px;
    margin-bottom: 15px;
    text-transform: capitalize;
  }
  
  
  .snippet{
    margin-bottom: 20px;
  }
  
  .container{
    margin-top: 30px;
    padding: 10px 10px;
  }
  
  