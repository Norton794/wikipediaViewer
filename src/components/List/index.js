import PropTypes from "prop-types";
import './index.css';
export const List = ({ list }) => {
    return (
      <div className="container">
        {list &&
          list.query.search.map((item) => (
            <div className="cards" key={item.pageid}>
              <a href={`https://en.wikipedia.org/?curid=${item.pageid}`}>
                <h3 className="pgTitle">{item.title}</h3>
                <div
                  className="snippet"
                  dangerouslySetInnerHTML={{ __html: `<p>${item.snippet}</p>` }}
                />
              </a>
            </div>
          ))}
      </div>
    );
  };
  
  List.propTypes = {
    list: PropTypes.object,
  };