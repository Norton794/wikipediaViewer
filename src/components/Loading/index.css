/*Loading*/


.ball-one{
    margin-top: 30px;
      position: relative;
      width: 60px;
      height: 60px;
      background-color: #fff;
      border-radius: 50%;
      box-shadow: 2px 2px 3px 2px rgba(0,0,0,0.2);
      display: flex;
      justify-content: center;
      align-items: center;
      animation: ball-one 2s cubic-bezier(.21,.98,.6,.99) infinite alternate;
  }
  
  .ball-two{
      position: relative;
      width: 40px;
      height: 40px;
      background-color: #fff;
      border-radius: 50%;
      box-shadow: 2px 2px 3px 2px rgba(0,0,0,0.2);
      display: flex;
      justify-content: center;
      align-items: center;
      animation: ball-two 2s cubic-bezier(.21,.98,.6,.99) infinite alternate;
  }
  
  .ball-three{
      position: relative;
      width: 20px;
      height: 20px;
      background-color: #fff;
      border-radius: 50%;
      box-shadow: 2px 2px 3px 2px rgba(0,0,0,0.2);
      animation: ball-three 2s cubic-bezier(.21,.98,.6,.99) infinite alternate;
  }
  
  
  @keyframes ball-one {
      0% {
          transform: scale(0);
      }
  
      100% {
          box-shadow: 10px 10px 15px 0 rgba(0,0,0,0.3);
          transform: scale(1);
      }
  }
  
  @keyframes ball-two {
      0% {
          transform: scale(0);
      }
      
      40%{
          transform: scale(0);
      }
  
      100% {
          box-shadow: 10px 10px 15px 0 rgba(0,0,0,0.3);
          transform: scale(1);
      }
  }
  
  
  @keyframes ball-three {
      0% {
          transform: scale(0);
      }
      
      70%{
          transform: scale(0);
      }
  
      100% {
          box-shadow: 10px 10px 15px 0 rgba(0,0,0,0.3);
          transform: scale(1);
      }
  }