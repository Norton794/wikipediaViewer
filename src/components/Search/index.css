input{
    outline: none;
     border: none;
     border-radius: 5px;
     font-size: 16px;
     padding: 15px 15px;
     box-shadow: 2px 2px 2px 2px rgba(0, 0, 0, 0.2);
     width: 450px;
   }