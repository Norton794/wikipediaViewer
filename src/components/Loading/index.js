import {List} from '../../components/List/index';
import './index.css';
export const Loading = () => (
    <div className="ball-one">
      <div className="ball-two">
        <div className="ball-three"></div>
      </div>
    </div>
  );
  
  export const withLoading =
    (Component) =>
    ({ isLoading, ...rest }) =>
      isLoading ? <Loading /> : <Component {...rest} />;
  
  export const ListWithLoading = withLoading(List);