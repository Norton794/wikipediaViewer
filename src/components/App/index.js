import React, { Component } from "react";
import axios from "axios";
import './index.css';
import {
  BASE,
  ACTION,
  LIST,
  SEARCH,
  DEFAULT_SEARCH,
  FORMAT,
  ORIGIN,
  RANDOM,
} from "../../constants/index";

import {Search} from '../../components/Search/index';

import {ListWithLoading} from '../../components/Loading/index';

class App extends Component {
  _isMounted = false;
  constructor() {
    super();
    this.state = {
      result: null,
      searchTerm: DEFAULT_SEARCH,
      error: null,
      isLoading: false,
    };
    this.setSearchWiki = this.setSearchWiki.bind(this);
    this.fetchSearchWiki = this.fetchSearchWiki.bind(this);
    this.onSearchChange = this.onSearchChange.bind(this);
    this.onSearchSubmit = this.onSearchSubmit.bind(this);
  }

  setSearchWiki(result) {
    this.setState({ result, isLoading: false });
  }

  onSearchChange(e) {
    this.setState({ searchTerm: e.target.value });
  }

  fetchSearchWiki(searchTerm) {
    this.setState({ isLoading: true });

    axios(
      `${BASE}?${ACTION}&${LIST}&${SEARCH}${searchTerm}&${FORMAT}&${ORIGIN}`
    )
      .then((result) => this._isMounted && this.setSearchWiki(result.data))
      .catch((error) => this._isMounted && this.setState({ error }));
  }

  onSearchSubmit(e) {
    e.preventDefault();
    const { searchTerm } = this.state;
    searchTerm && this.fetchSearchWiki(searchTerm);
  }

  componentDidMount() {
    this._isMounted = true;

    const { searchTerm } = this.state;

    this.fetchSearchWiki(searchTerm);
  }

  componentWillUnmount() {
    this._isMounted = false;
  }

  render() {
    const { result, searchTerm, error, isLoading } = this.state;

    return (
      <div className="page">
        <h1>WikiViewer</h1>

        <a className="random" href={RANDOM}>
          Go to a random article
        </a>

        <Search
          onChange={this.onSearchChange}
          value={searchTerm}
          onSubmit={this.onSearchSubmit}
        ></Search>
        {error ? 
          <p className="error">Something went wrong :(</p>
         : <ListWithLoading list={result} isLoading={isLoading}/>
        }
      </div>
    );
  }
}





export default App;

