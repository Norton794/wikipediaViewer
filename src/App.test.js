import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import App from "../src/components/App/index";
import { Search } from "../src/components/Search/index";
import { List } from "../src/components/List/index";
import { Loading } from "../src/components/Loading/index";

Enzyme.configure({ adapter: new Adapter() });

describe("App", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test("has a valid snapshot", () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Search", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Search>Search</Search>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test("has a valid snapshot", () => {
    const component = renderer.create(<Search>Search</Search>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("List", () => {
  const props = {
    list: {
      query: {
        search: [
          {
            ns: 0,
            title: "Deadpool and Korg React",
            pageid: 68266160,
            size: 18176,
            wordcount: 1370,
            snippet:
              'Deadpool and Korg <span class="searchmatch">React</span> is a 2021 American superhero promotional short film featuring the Marvel Comics characters Deadpool and Korg. The film was directed',
            timestamp: "2021-10-09T03:52:37Z",
          },
          {
            ns: 0,
            title: "React Quotes",
            pageid: 14985089,
            size: 8993,
            wordcount: 861,
            snippet:
              '&quot;<span class="searchmatch">React</span> Quotes&quot; is the fifth episode of the fifth season of the HBO original series, The Wire. The episode was written by David Mills from a story by David',
            timestamp: "2021-05-30T04:35:31Z",
          },
          {
            ns: 0,
            title: "React to That",
            pageid: 43361791,
            size: 9355,
            wordcount: 323,
            snippet:
              '<span class="searchmatch">React</span> to That (stylized as <span class="searchmatch">React</span>ToThat) is an American television series developed by Nick Cannon and Benny and Rafi Fine. It aired on Nickelodeon for',
            timestamp: "2021-07-03T21:26:24Z",
          },
        ],
      },
    },
  };

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<List {...props} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test("has a valid snapshot", () => {
    const component = renderer.create(<List {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("shows two items in list", () => {
    const element = shallow(<List {...props} />);

    expect(element.find(".cards").length).toBe(3);
  });
});



describe("Loading", () => {

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Loading />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test("has a valid snapshot", () => {
    const component = renderer.create(<Loading />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});