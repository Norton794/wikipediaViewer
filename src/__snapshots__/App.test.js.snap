// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has a valid snapshot 1`] = `
<div
  className="page"
>
  <h1>
    WikiViewer
  </h1>
  <a
    className="random"
    href="https://en.wikipedia.org/wiki/Special:Random"
  >
    Go to a random article
  </a>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Search with the Wikipedia API"
      type="text"
      value="React"
    />
  </form>
  <div
    className="ball-one"
  >
    <div
      className="ball-two"
    >
      <div
        className="ball-three"
      />
    </div>
  </div>
</div>
`;

exports[`List has a valid snapshot 1`] = `
<div
  className="container"
>
  <div
    className="cards"
  >
    <a
      href="https://en.wikipedia.org/?curid=68266160"
    >
      <h3
        className="pgTitle"
      >
        Deadpool and Korg React
      </h3>
      <div
        className="snippet"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Deadpool and Korg <span class=\\"searchmatch\\">React</span> is a 2021 American superhero promotional short film featuring the Marvel Comics characters Deadpool and Korg. The film was directed</p>",
          }
        }
      />
    </a>
  </div>
  <div
    className="cards"
  >
    <a
      href="https://en.wikipedia.org/?curid=14985089"
    >
      <h3
        className="pgTitle"
      >
        React Quotes
      </h3>
      <div
        className="snippet"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>&quot;<span class=\\"searchmatch\\">React</span> Quotes&quot; is the fifth episode of the fifth season of the HBO original series, The Wire. The episode was written by David Mills from a story by David</p>",
          }
        }
      />
    </a>
  </div>
  <div
    className="cards"
  >
    <a
      href="https://en.wikipedia.org/?curid=43361791"
    >
      <h3
        className="pgTitle"
      >
        React to That
      </h3>
      <div
        className="snippet"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p><span class=\\"searchmatch\\">React</span> to That (stylized as <span class=\\"searchmatch\\">React</span>ToThat) is an American television series developed by Nick Cannon and Benny and Rafi Fine. It aired on Nickelodeon for</p>",
          }
        }
      />
    </a>
  </div>
</div>
`;

exports[`Loading has a valid snapshot 1`] = `
<div
  className="ball-one"
>
  <div
    className="ball-two"
  >
    <div
      className="ball-three"
    />
  </div>
</div>
`;

exports[`Search has a valid snapshot 1`] = `
<form>
  <input
    placeholder="Search"
    type="text"
  />
</form>
`;
